core/src/main/java/mod/chiselsandbits/inventory/bit/SlottedBitInventoryItemStack.java

```diff
@@ -70,6 +70,55 @@ public void clear(final IBlockInformation state)
             }
         }
     }
+    @Override
+    public void convert(Player player) {
+        // Get counts of all the bits present in the bag and clear it.
+        final Map<IBlockInformation, Integer> contentMap = Maps.newHashMap();
+        this.slotMap.values().forEach(bitSlot -> {
+                    contentMap.putIfAbsent(bitSlot.getBlockInformation(), 0);
+                    contentMap.compute(bitSlot.getBlockInformation(), (s, c) -> (c == null ? 0 : c) + bitSlot.getCount());
+                }
+        );
+        this.slotMap.clear();
+
+
+
+        List<Map.Entry<IBlockInformation, Integer>> toSort = new ArrayList<>(contentMap.entrySet());
+        toSort.sort(Map.Entry.<IBlockInformation, Integer>comparingByValue().reversed());
+
+        int slotIndex = 0;
+        for (Map.Entry<IBlockInformation, Integer> e : toSort)
+        {
+            int count = e.getValue();
+            if (count == 0) {
+                continue;
+            }
+
+            // Give player items for each 4096 bits they have (full block) 16^3
+            while (count >= 4096) {
```

Comment:
The current state entry size needs to be questioned to check for how many bits are in a block.

Reply:
Done! We are getting that number from there now.